### PR TITLE
docs/managed: adds alertmanager configuration examples

### PR DESCRIPTION
**Pull request by f41gh7**

### CODE REVIEW

File: docs/managed-victoriametrics/alertmanager_setup_for_deployment.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+* `victorops_configs`
+* `telegram_configs`
+* `webex_configs`
+* `msteams_configs`
+
+### forbidden keys
+
+ All configuration params with `_file` suffix is not allowed for security reasons.
+
+### reusable templates
```
Comment (f41gh7) [inline docs/managed-victoriametrics/alertmanager_setup_for_deployment.md:37]: remove for now

File: docs/managed-victoriametrics/alertmanager_setup_for_deployment.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+- name: dev-team-2
+  slack_configs:
+  - api_url: https://hooks.slack.com/services/valid-url
+    channel: dev-alerts
+{% endraw %}
+```
+
+### Configure alerting rules
+ Alerting and recording rules configured via API calls.
+ With first uploaded rule file, alerting stack is started.
```
Comment (f41gh7) [inline docs/managed-victoriametrics/alertmanager_setup_for_deployment.md:130]: remove details

**Review by tenmozes — APPROVED** [review]

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
+---
+title: Alertmanager and VMAlert configuration for Deployment
+aliases:
+- /managed-victoriametrics/alertmanager-configuration.html
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:4]: no need that. for every new link we don't need redicrect

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+* `msteams_configs`
+
+### forbidden keys
+
+ All configuration params with `_file` suffix is not allowed for security reasons.
+
+### Configuration example
+
+```yaml
+{% raw %}
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:40]: no need {% raw %}

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    channel: dev-alerts
+- name: dev-team-2
+  slack_configs:
+  - api_url: https://hooks.slack.com/services/valid-url
+    channel: dev-alerts
+{% endraw %}
+```
+
+## Configure alerting rules
+ Alerting and recording rules could be configured via API calls.
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:102]: space after title (applicable for all )

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+title: Alertmanager and VMAlert configuration for Deployment
+aliases:
+- /managed-victoriametrics/alertmanager-configuration.html
+---
+
+## Alerting stack configuration and Managed VictoriaMetrics
+
+ Managed VictoriaMetrics supports configuration for alerting rules and notifications for it with alertmanager.
+
+## Configure alertmanager
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:11]: Alertmanager (pleae change everywhere)

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+* `webhook_configs`
+* `opsgenie_configs`
+* `wechat_configs`
+* `pushover_configs`
+* `victorops_configs`
+* `telegram_configs`
+* `webex_configs`
+* `msteams_configs`
+
+### forbidden keys
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:33]: forbidden keys -> Limitation

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+ Alerting and recording rules could be configured via API calls.
+
+### Managed VictoriaMetrics rules API
+
+ Managed VictoriaMetrics has following APIs for rules:
+
+* POST: `/api/v1/deployments/{deploymentId}/rule-sets/files/{fileName}`
+* DELETE `/api/v1/deployments/{deploymentId}/rule-sets/files/{fileName}`
+
+ Swagger API examples [link](https://cloud.victoriametrics.com/api-docs)
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:111]: Swagger -> OpenAPI

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+### rules creation with API
+
+Lets create a 2 simple rules for deployment at `testing-rules.yaml`
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:115]: Let's create two example rules for deployment in testing-rules.yaml

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+      - alert: always-firing
+        expr: vector(1) > 0 
+        for: 30s
+        labels:
+          severity: critical
+        annotations:
+          summary: "rule must be always at firing state"
+```
+
+ Upload rules into the Managed VictoriaMetrics with following command:
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:139]: Upload rules to the Managed VictoriaMetrics using the following command:

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+
+```sh
+curl https://https://cloud.victoriametrics.com/api/v1/deployments/DEPLOYMENT_ID/rule-sets/files/testing-rules -v -H 'X-VM-Cloud-Access: CLOUD_API_TOKEN' -XPOST --data-binary '@testing-rules.yaml'
+```
+
+## Troubleshooting
+
+### rules state check
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:147]: please rephrase i don't follow the title

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 12]
```diff
@@ -0,0 +1,165 @@
+---
+title: Alertmanager and VMAlert configuration for Deployment
+aliases:
+- /managed-victoriametrics/alertmanager-configuration.html
+---
+
+## Alerting stack configuration and Managed VictoriaMetrics
+
+ Managed VictoriaMetrics supports configuration for alerting rules and notifications for it with alertmanager.
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:9]: Managed VictoriaMetrics supports configuring alerting rules and notifications through Alertmanager and internal vmalert.

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+
+## Configure alertmanager
+
+ Managed VictoriaMetrics supports alertmanager with standard [configuration](https://prometheus.io/docs/alerting/latest/configuration/).
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:13]: Managed VictoriaMetrics supports Alertmanager with a standard [configuration](https://prometheus.io/docs/alerting/latest/configuration/). The configuration menu is located on the deployment page under the Alertmanager section.

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+Configuration menu located at `deployment` page under `Alertmanager` section.
+
+<img src="alertmanager_location.webp">
+ Configuration parameters have following limitations:
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:17]: please check the configuration options and limitations

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 15]
```diff
@@ -0,0 +1,165 @@
...
+* `wechat_configs`
+* `pushover_configs`
+* `victorops_configs`
+* `telegram_configs`
+* `webex_configs`
+* `msteams_configs`
+
+### forbidden keys
+
+ All configuration params with `_file` suffix is not allowed for security reasons.
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:35]: All configuration parameters with a _file suffix are not allowed for security reasons.

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md (hunk 5 above)
Comment (tenmozes) [reply docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:102]: Alerting and recording rules can be configured via API calls.

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+
+```sh
+curl https://https://cloud.victoriametrics.com/api/v1/deployments/DEPLOYMENT_ID/rule-sets/files/testing-rules -v -H 'X-VM-Cloud-Access: CLOUD_API_TOKEN' -XPOST --data-binary '@testing-rules.yaml'
+```
+
+## Troubleshooting
+
+### rules state check
+
+ Created rules state located at `rules` section for Deployment:
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:149]: The state of created rules is located in the rules section for Deployment:

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+
+<img src="alertmanager_rules_state.webp">
+
+### alerts state check
+
+ Alerts and silences could be found at alertmanager section.
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:155]: Alerts and silences can be found in the Alertmanager section.

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+
+<img src="alertmanager_alerts_state.webp">
+
+### debug
+
+ It's possible to debug alerting stack with logs for `vmalert` and `alertmanager` exposed at `Logs` section of deployment.
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:161]: It's possible to debug the alerting stack with logs for vmalert and alertmanager, which are accessible in the Logs section of the deployment.

File: docs/managed-victoriametrics/alertmanager-setup-for-deployment.md [hunk 19]
```diff
@@ -0,0 +1,165 @@
...
+
+<img src="alertmanager_troubleshoot_logs.webp">
+
+ Alertmanager integration errors are also tracked by internal cloud monitoring system.
```
Comment (tenmozes) [inline docs/managed-victoriametrics/alertmanager-setup-for-deployment.md:165]: Please provide information what it means and why this is important, eg what guarantees we have or what expect from us

**Review by dmitryk-dk — APPROVED** [review]
LGTM!

**Pull request merged**